pkg/uninstaller/resources.go

```diff
@@ -0,0 +1,198 @@
+package uninstaller
+
+import (
+	"context"
+	clusterconfigV1alpha1 "github.com/liqotech/liqo/apis/config/v1alpha1"
+	discoveryV1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	"github.com/liqotech/liqo/apis/net/v1alpha1"
+	advertisementOperator "github.com/liqotech/liqo/pkg/advertisement-operator"
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog/v2"
+	"sync"
+	"time"
+)
+
+type toCheckDeleted struct {
+	gvr           schema.GroupVersionResource
+	labelSelector metav1.LabelSelector
+}
+
+type resultType struct {
+	Resource toCheckDeleted
+	Success  bool
+}
+
+var (
+	podGVR = v1.SchemeGroupVersion.WithResource("pods")
+
+	toCheck = []toCheckDeleted{
+		{
+			gvr:           v1alpha1.TunnelEndpointGroupVersionResource,
+			labelSelector: metav1.LabelSelector{},
+		},
+		{
+			gvr:           v1alpha1.NetworkConfigGroupVersionResource,
+			labelSelector: metav1.LabelSelector{},
+		},
+		{
+			gvr: podGVR,
+			labelSelector: metav1.LabelSelector{
+				MatchLabels: advertisementOperator.KubeletBaseLabels,
+			},
+		},
+		{
+			gvr: podGVR,
+			labelSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": "broadcaster"},
+			},
+		},
+	}
+)
+
+func UnjoinClusters(client dynamic.Interface) error {
+	r1 := client.Resource(discoveryV1alpha1.ForeignClusterGroupVersionResource)
+	t, err := r1.Namespace("").List(context.TODO(), metav1.ListOptions{TypeMeta: metav1.TypeMeta{}})
+	if err != nil {
+		return err
+	}
+	klog.V(8).Info("Getting ForeignClusters list")
```

Comment:
I would use a verbosity lower than 8, which is very high. I think that if you set 8, finding what you need without grepping can be very difficult. Moreover, consider that the program is fairly simple, we don't have too many prints.